builtin/logical/aws/secret_access_keys.go

```diff
@@ -67,14 +68,26 @@ func genUsername(displayName, policyName, userType string) (ret string, warning
 func (b *backend) secretTokenCreate(ctx context.Context, s logical.Storage,
 	displayName, policyName, policy string,
 	lifeTimeInSeconds int64) (*logical.Response, error) {
-	STSClient, err := clientSTS(ctx, s)
-	if err != nil {
-		return logical.ErrorResponse(err.Error()), nil
+
+	b.clientMutex.RLock()
```

Comment:
May it be a good idea to abstract out the nil checking and client setting in the backend struct? I see that all the functions that use either `b.iamClient` or `b.stsClient` should check those for `nil` and perform the logic of setting it in the backend. This seems easy to get wrong, or simply forget.

How about we push this logic in a function, say `b.IAMClient()` and `b.STSClient()` which return the `b.iamClient` and `b.stsClient` respectively if populated. If not, it can then do the locking to set the client and return the client. The calling functions will then have to use the *returned* client as opposed to `b.[iam|sts]Client`. They way calling functions can be sure that the client is a non-nil value.

Reply:
Seems reasonable, I made the changes in 3c74986d2 please take a look